nf_core/components/components_command.py

```diff
@@ -0,0 +1,195 @@
+        """
+        Get the local modules/subworkflows in a pipeline
+        """
+        local_component_dir = Path(self.dir, self.component_type, "local")
+        return [
+            str(path.relative_to(local_component_dir)) for path in local_component_dir.iterdir() if path.suffix == ".nf"
+        ]
+
+    def get_components_clone_modules(self):
+        """
+        Get the modules/subworkflows repository available in a clone of nf-core/modules
+        """
+        if self.component_type == "modules":
+            component_base_path = Path(self.dir, self.default_modules_path)
+        elif self.component_type == "subworkflows":
+            component_base_path = Path(self.dir, self.default_subworkflows_path)
+        return [
+            str(Path(dir).relative_to(component_base_path))
+            for dir, _, files in os.walk(component_base_path)
+            if "main.nf" in files
+        ]
+
+    def has_valid_directory(self):
+        """Check that we were given a pipeline or clone of nf-core/modules"""
+        if self.repo_type == "modules":
+            return True
+        if self.dir is None or not os.path.exists(self.dir):
+            log.error(f"Could not find pipeline: {self.dir}")
+            return False
+        main_nf = os.path.join(self.dir, "main.nf")
+        nf_config = os.path.join(self.dir, "nextflow.config")
+        if not os.path.exists(main_nf) and not os.path.exists(nf_config):
+            if Path(self.dir).resolve().parts[-1].startswith("nf-core"):
+                raise UserWarning(f"Could not find a 'main.nf' or 'nextflow.config' file in '{self.dir}'")
+            log.warning(f"Could not find a 'main.nf' or 'nextflow.config' file in '{self.dir}'")
+        return True
+
+    def has_modules_file(self):
+        """Checks whether a module.json file has been created and creates one if it is missing"""
+        modules_json_path = os.path.join(self.dir, "modules.json")
+        if not os.path.exists(modules_json_path):
+            log.info("Creating missing 'module.json' file.")
+            ModulesJson(self.dir).create()
+
+    def clear_module_dir(self, module_name, module_dir):
+        """Removes all files in the module directory"""
+        try:
+            shutil.rmtree(module_dir)
+            # Try cleaning up empty parent if tool/subtool and tool/ is empty
+            if module_name.count("/") > 0:
+                parent_dir = os.path.dirname(module_dir)
+                try:
+                    os.rmdir(parent_dir)
+                except OSError:
+                    log.debug(f"Parent directory not empty: '{parent_dir}'")
+                else:
+                    log.debug(f"Deleted orphan tool directory: '{parent_dir}'")
+            log.debug(f"Successfully removed {module_name} module")
+            return True
+        except OSError as e:
+            log.error(f"Could not remove module: {e}")
+            return False
+
+    def modules_from_repo(self, install_dir):
```

Comment:
Rename this method to reflect its dual capability?

```suggestion
    def components_from_repo(self, install_dir):
```